internal/formats/common/cyclonedxhelpers/external_references.go

```diff
@@ -4,12 +4,16 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/anchore/syft/internal/file"
+	syftFile "github.com/anchore/syft/syft/file"
+
 	"github.com/CycloneDX/cyclonedx-go"
 	"github.com/anchore/syft/syft/pkg"
 )
 
+//nolint:funlen
 func encodeExternalReferences(p pkg.Package) *[]cyclonedx.ExternalReference {
-	refs := []cyclonedx.ExternalReference{}
+	var refs []cyclonedx.ExternalReference
```

Comment:
Will this possibly result in some `nil` pointer issue later?

Reply:
You can still append to and assert the length of a nil slice. Goland suggested moving from a literal to a nil slice declaration in this case.

Here is a good StackOverflow on how nil slice declarations behave:
https://stackoverflow.com/questions/44305170/nil-slices-vs-non-nil-slices-vs-empty-slices-in-go-language